addons/web_tour/static/src/js/tip.js

```diff
@@ -156,41 +235,72 @@ var Tip = Widget.extend({
+
+        const center = (paddingLeft + paddingRight) + ((($parent[0].clientWidth - (paddingLeft + paddingRight)) / 2) - this.$el[0].offsetWidth / 2);
+        const topPosition = $parent[0].offsetTop;
+        let top;
+        if (this.viewPortState === 'up') {
+            top = topPosition + this.$el.innerHeight() + paddingTop;
+            this.$el.css({top: top, left: center});
+        } else if (this.viewPortState === 'down') {
+            top = topPosition + $parent.innerHeight() - this.$el.innerHeight() * 2;
+            this.$el.css({top: top, left: center});
+        } else {
+            this.$el.position({
+                my: this._get_spaced_inverted_position(appendAt),
+                at: appendAt,
+                of: this.$anchor,
+                collision: "none",
+            });
+        }
 
         // Reverse overlay if direction is right to left
         var positionRight = _t.database.parameters.direction === 'rtl' ? "right" : "left";
         var positionLeft = _t.database.parameters.direction === 'rtl' ? "left" : "right";
-        var offset = this.$el.offset();
+
+        // get the offset position of this.$el
+        // Couldn't use offset() or position() because their values are not the desired ones in all cases
+        const offset = {top: this.$el[0].offsetTop, left: this.$el[0].offsetLeft};
         this.$tooltip_overlay.css({
             top: -Math.min((this.info.position === "bottom" ? this.info.space : this.info.overlay.y), offset.top),
             right: -Math.min((this.info.position === positionRight ? this.info.space : this.info.overlay.x), this.$window.width() - (offset.left + this.init_width + this.double_border_width)),
             bottom: -Math.min((this.info.position === "top" ? this.info.space : this.info.overlay.y), this.$window.height() - (offset.top + this.init_height + this.double_border_width)),
             left: -Math.min((this.info.position === positionLeft ? this.info.space : this.info.overlay.x), offset.left),
         });
-
-        this.position = this.$el.position();
+        this.position = offset;
 
         this.$el.addClass("o_animated");
     },
     _bind_anchor_events: function () {
-        this.consume_event = Tip.getConsumeEventType(this.$anchor);
+        this.consume_event = this.info.consumeAction ? this.info.consumeAction : Tip.getConsumeEventType(this.$anchor);
         this.$consumeEventAnchor = this.$anchor;
         // jQuery-ui draggable triggers 'drag' events on the .ui-draggable element,
         // but the tip is attached to the .ui-draggable-handle element which may
         // be one of its children (or the element itself)
         if (this.consume_event === "drag") {
             this.$consumeEventAnchor = this.$anchor.closest('.ui-draggable');
+        } else if (this.consume_event === "input") {
+            this.initialContent = this.$anchor.text().trim();
+            this.$consumeEventAnchor = this.$anchor.closest("[contenteditable='true']")
```

Comment:
Right, maybe the tour wrote something in every cases ? . But yes, I could do something to chose which steps should be consumed by an input when there is text to add.

Reply:
I don't understand what you said. Here you say "oh it's an input event which consumes this tip? ok let's find the contenteditable element"... what about the normal `<input/>`, `<textarea/>` which also use consume_event?

Reply:
What's the goal, to consume the tip for every `input`, `textarea` and `contenteditable` element with an input ? Even the one that aren't mandatory ? 
Does every `input` and `textarea` are inside a `contenteditable` div ? I don't really know every cases and don't sure how to set it up correctly :/.
In this case I'm just wait an input by the user when `consume_event === input`. It has this value in 2 situations, or it's explicitly write in the step or it uses `getConsumeEventType()`.

Reply:
Please look at the code before your PR and the code now, I don't know how to explain this to you more clearly... Before your PR, "input" consume_event in `<input>` and `<textarea>` were listened on those element themself, now with your code it will search for a contenteditable parent element which we would not find in those normal cases.....